docs/data-structures/palindromic-linked-list.mdx

```diff
@@ -0,0 +1,158 @@
+   Remember that this data is actually the data of the last node of the linked list.
+6. Compare the value in the temporary variable with that of the node pointed to by the pointer in the linked list
+   Remember that the pointer points to the head , that is, the first node of the linke dlist.
+7. If the two values are the same, then move onto the next node in the linked list and pop the current top element in the stack.
+   Repeat step 6 till the end of the stack or linked list is reached or the two values being compared are not the same.
+8. If the end of the linked list is reached, then it means that the linked list is palindromic. Otherwise it is not palindromic.
+
+# Examples
+
+## Example 1
+
+Let the linked list be 1->2->3
+All the data is pushed into the stack in the order: 3 2 1 with 1 at the bottom and 3 at top
+The pointer points towards the head of the linked list that is 1
+It is now compared with the top of the stack which is 3, so they are not equal and hence linked list is not palindrome.
+
+## Example 2
+
+Let the linked list be 1->2->1
+All the data is pushed into the stack 
+Compare the top of the stack(=1) with the data at head of the linked list.
+As they are equal, pop the current top of the stack and pointer points towards the next node(=2)
+Again both are equal. Repeat the above step again. 
+You have reached the end of the linked list. therefore, it is palindromic.
+
+
+# Code
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+<Tabs
+  defaultValue="cpp"
+  values={[
+    { label: 'C++', value: 'cpp', }
+  ]
+}>
+
+<TabItem value="cpp">
+
+```cpp
+
+#include <bits/stdc++.h>
+using namespace std;
+struct Node
+{
+    int data;
+    Node *next;
+};
+void push(Node **head, int new_data)
+{
+    Node *temp = new Node();
+    temp->data = new_data;
+    temp->next = *head;
+    *head = temp;
+}
+int checkPalin(Node *head)
+{
+    stack<int> s;
+    Node *temp = head;
+    while (temp != NULL)
+    {
+        s.push(temp->data);
+        temp = temp->next;
+    }
+    Node *ref = new Node();
+    ref = head;
+    while (ref != NULL)
+    {
+        int i = s.top();
+        s.pop();
+        if (i != ref->data)
+        {
+            return -1;
+        }
+        ref = ref->next;
+    }
+    return 1;
+}
+
+//Driver code
+int main()
+{
+    int n;
+    cout<<"Enter the length of the list ";
+    cin>>n;
+    int a[n];
+    cin>>a[0];
+    Node *head = new Node();
+    head->data = a[0];
+    head->next = NULL;
+    cout<<"\nEnter the data\n";
+    for(int i=1;i<n;i++)
+    {
+        cin>>a[i];
+        push(&head,a[i]);
+    }
+    if(checkPalin == 1)
+    cout << "Yes, linked list is palindrome";
+    else if( checkPalin == -1)
+    cout<< "No, linked list is not palindrome";
+}
+
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+## Sample 1
+
+### Input:
+Enter the length of the list 3
+Enter the data
+1
+2
+1
+
+### Output:
+Yes,linked list is palindrome
```

Comment:
Interesting, your print statement has whitespace, never the less, pls add one here as well.

Reply:
Done